src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -887,7 +887,7 @@ public BoundCall Call(BoundExpression? receiver, MethodSymbol method, ImmutableA
             return new BoundCall(
                 Syntax, receiver, initialBindingReceiverIsSubjectToCloning: ThreeState.Unknown, method, args,
                 argumentNamesOpt: default(ImmutableArray<String?>), argumentRefKindsOpt: refKinds, isDelegateCall: false, expanded: false, invokedAsExtensionMethod: false,
-                argsToParamsOpt: ImmutableArray<int>.Empty, defaultArguments: default(BitVector), resultKind: LookupResultKind.Viable, type: method.ReturnType)
+                argsToParamsOpt: default, defaultArguments: default(BitVector), resultKind: LookupResultKind.Viable, type: method.ReturnType)
```

Comment:
>argsToParamsOpt: default, [](http://example.com/codeflow?start=16&length=25)

Why is this change necessary?

Reply:
Because the utility `Binder.GetCorrespondingParameter` that's used in this PR as part of `CodeGenerator.MightEscapeTemporaryRefs` would fail on an assert - it expects `argsToParamsOpt` to be either default or matching the number of arguments:

https://github.com/dotnet/roslyn/blob/14244f6fcdf77663d7e866ebcdf45e6881a51195/src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs#L1447

Reply:
Is this change still necessary? I.e. is anything going to fail if it is reverted? If nothing is going to fail, I suggest unbundling it from the fix this PR implements.